particle/particle/utilities.py

```diff
@@ -64,3 +65,74 @@ def str_with_unc(value, upper, lower=None):
         return '{value:{fsv}} + {upper:{fse}} - {lower:{fse}}'.format(value=value,
                                                                       upper=upper, lower=lower,
                                                                       fsv=fsv, fse=fse)
+
+list_name_greek_letters = (
+    'Alpha',
+    'Beta',
+    'Chi',
+    'Delta',
+    'Epsilon',
+    'Eta',
+    'Gamma',
+    'Iota',
+    'Kappa',
+    'Lambda',
+    'Mu',
+    'Nu',
+    'Omega',
+    'Omicron',
+    'Phi',
+    'Pi',
+    'Psi',
+    'Rho',
+    'Sigma',
+    'Tau',
+    'Theta',
+    'Upsilon',
+    'Xi',
+    'Zeta',
+    'alpha',
+    'beta',
+    'chi',
+    'delta',
+    'epsilon',
+    'eta',
+    'gamma',
+    'iota',
+    'kappa',
+    'lambda',
+    'mu',
+    'nu',
+    'omega',
+    'omicron',
+    'phi',
+    'pi',
+    'psi',
+    'rho',
+    'sigma',
+    'tau',
+    'theta',
+    'upsilon',
+    'xi',
+    'zeta'
+)
+
+
+def greek_letter_name_to_unicode(letter):
+    if letter == letter.lower():
+        return unicodedata.lookup('GREEK SMALL LETTER %s' % letter.upper())
+    else:
+        return unicodedata.lookup('GREEK CAPITAL LETTER %s' % letter.upper())
+
+
+def latex_to_html_name(name):
+    """Conversion of particle names from LaTeX to HTML."""
+    name = re.sub(r'\^\{(.*?)\}', r'<SUP>\1</SUP>', name)
+    name = re.sub(r'\_\{(.*?)\}', r'<SUB>\1</SUB>', name)
+    name = re.sub(r'\\mathrm\{(.*?)\}', r'\1', name)
+    name = re.sub(r'\\left\[(.*?)\\right\]', r'[\1] ', name)
+    for gl in list_name_greek_letters:
+        # Special care - unicodedata library uses "lamda" for "lambda" :S!
```

Comment:
Couldn't you first replace lambda with lamda, then try to convert to unicode name? It would be nice to avoid the explicit listing.

Reply:
You mean replacing
```
...
name = name.replace(r'\%s'%gl, greek_letter_name_to_unicode(gl.replace('ambda', 'amda')))
```
with
```
... 
name = name.replace('ambda', 'amda')
...
name = name.replace(r'\%s'%gl, greek_letter_name_to_unicode(gl))
```
?

Reply:
Yes, esp. if that allows you to avoid the list and just use a try catch block.

Reply:
Something like:

```python
try:
    unicodedata.lookup('GREEK SMALL LETTER MATHRM')
except KeyError:
    return unchanged
```

will return `unchanged`.

Reply:
I'm not sure about the above. In any case I've just committed a simplified version of the LaTeX -> HTML conversion code.

Reply:
This would have been done a different way; you'd look for all `/<letters>` and then try to convert them. It's a larger change; let's go with this for now.